src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainParameters.cs

```diff
@@ -9,6 +9,76 @@ namespace Nethermind.Specs.ChainSpecStyle;
 
 public class ChainParameters
 {
+
+    private ulong? _dencunTransitionTimestamp;
+    private ulong? _cancunTransitionTimestamp;
+
+    public ulong? DencunTransitionTimestamp
+    {
+        get
+        {
+            if (Eip4844TransitionTimestamp == Eip4788TransitionTimestamp &&
+                Eip4788TransitionTimestamp == Eip1153TransitionTimestamp &&
+                Eip1153TransitionTimestamp == Eip5656TransitionTimestamp &&
+                Eip5656TransitionTimestamp == Eip6780TransitionTimestamp)
+            {
+                return Eip4844TransitionTimestamp;
+            }
+
+            return null;
+        }
+        set
+        {
+            _dencunTransitionTimestamp = value;
```

Comment:
Potentially we could have `_dencunTransitionTimestamp` and individual EIP's out of sync, so maybe always calculate it and don't have a separate field? Then you also don't need `Validate` methods